src/collation.ts

```diff
@@ -182,3 +182,70 @@ export const codepageBySortId: { [key: number]: string | undefined } = {
   [185]: 'CP1252', // SQL_SwedishStd_Pref_Cp1_CI_AS_KI_WI
   [186]: 'CP1252' // SQL_Icelandic_Pref_Cp1_CI_AS_KI_WI
 };
+
+export const Flags = {
+  IGNORE_CASE: 1 << 0,
+  IGNORE_ACCENT: 1 << 1,
+  IGNORE_KANA: 1 << 2,
+  IGNORE_WIDTH: 1 << 3,
+  BINARY: 1 << 4,
+  BINARY2: 1 << 5,
+  UTF8: 1 << 6,
+};
+
+export class Collation {
+  readonly lcid: number;
+  readonly flags: number;
+  readonly version: number;
+  readonly sortId: number;
+  readonly codepage: string;
+
+  private buffer: Buffer | undefined;
+
+  static fromBuffer(buffer: Buffer, offset = 0) {
+    let lcid = (buffer[offset + 2] & 0x0F) << 16;
+    lcid |= buffer[offset + 1] << 8;
+    lcid |= buffer[offset + 0];
+
+    let flags = (buffer[offset + 3] & 0x0F) << 4;
+    flags |= (buffer[offset + 2] & 0xF0) >> 4;
```

Comment:
I suspect all our right shifts should either be unsigned (using `>>>` instead of `>>`) or careful to mask to the expected bitwidth to avoid sign extension turning on bits that shouldn’t be on.

Reply:
Done! Thanks for the suggestion. 🙇‍♂️